src/Fortran-interface/bml_getters_m.F90

```diff
@@ -117,6 +117,7 @@ subroutine bml_get_row_single_real(a, i, row)
     ptr = bml_get_row_C(a%ptr, i - 1)
     call c_f_pointer(ptr, row_ptr, [bml_get_N(a)])
     row = row_ptr
+    deallocate(row_ptr)
```

Comment:
This fixes the valgrind test. This suggests that the assignment in https://github.com/lanl/bml/pull/347/commits/39050c9f2e4c31b0af8a6aa595ef8c6401918115#diff-fc698c8987a3d4f990baad818a46a573R119 is copying the data instead of assigning a reference to it. @cnegre @jeanlucf22 @jmohdyusof @suemni what do you think?